packages/server-wallet/src/wallet/types.ts

```diff
@@ -60,6 +61,8 @@ export type ObjectiveResult = {
   channelId: string;
 };
 
-export type ObjectiveProposed = {
-  ObjectiveProposed: WalletObjective;
-};
+export interface WalletEvents {
+  ObjectiveCompleted: (o: WalletObjective) => void;
+  ObjectiveProposed: (o: WalletObjective) => void;
+  ObjectiveTimedOut: (o: WalletObjective) => void;
```

Comment:
The new `ObjectiveCompleted` and `ObjectiveTimedOut` are for internal use in the `wallet` but I think it's ok to expose them externally? (And easy to change if needed).